### DISCUSSION TITLE
Is it better for SEO to use 301 redirect instead of 302 one?

### DISCUSSION THREAD

**Discussion by annablgkv** [nuxt-modules/i18n#1307] [Q&A] [answered]
https://github.com/nuxt-community/i18n-module/blob/c710ad18d30dc80c9e71502fbf0e722e596952c7/src/templates/plugin.main.js#L209

**Comment by rchl** [comment, marked as answer]
Yes, I would say so.
The 301 can be very annoying when testing locally due to the redirect being cached but in production I would agree.

I guess your question is whether this default should be changed? Possibly but it would be a breaking change so I'm a little hesitant. At least it's customizable already.

**Reply by annablgkv** [reply]
```At least it's customizable already``` - can you show me how to do this customization? For now I'm writing my own middleware to change this redirect behavior only and it doesn't seem good at all(

**Reply by rchl** [reply]
It's in documentation https://i18n.nuxtjs.org/options-reference#rootredirect

**Reply by annablgkv** [reply]
what I really need to do is: redirect user from `/` to `/en` AND from `/my-custom-url` to `/en/my-custom-url` (where /en - is users browser language) with status code 301. that's why root redirect only doesn't help me a lot.
Do I miss something and this can be done with correct options in config?

**Reply by rchl** [reply]
Right. That option is only handling the root redirect.

The redirects to a detected locale do indeed use 302. Using 301 for those would be risky because the browser would cache the first redirect and what if the user would change the locale later? The browser would keep redirecting to the previous locale.

**Reply by annablgkv** [reply]
User can change language using language list menu.
But still, you're right: next visit without locale in the url will lead to a cached result, not the language that user has switched to previously. It's a pity, but SEO-specialist preffers 301 redirect in this case, also nextjs i18n works this way. 

Thank you a lot for help!
